lib/galaxy/webapps/galaxy/controllers/history.py

```
@@ -184,28 +184,29 @@ def make_private(self, trans, history_id=None, all_histories=False, **kwd):
         }
         for history in histories:
-            self.history_manager.error_unless_mutable(history)
-            # Set default role for history to private
-            trans.app.security_agent.history_set_default_permissions(history, private_permissions)
-            # Set private role for all datasets
-            for hda in history.datasets:
-                if (
-                    not hda.dataset.library_associations
-                    and not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset)
-                    and trans.app.security_agent.can_manage_dataset(user_roles, hda.dataset)
-                ):
-                    # If it's not private to me, and I can manage it, set fixed private permissions.
-                    trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)
-                    if not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset):
-                        raise exceptions.InternalServerError("An error occurred and the dataset is NOT private.")
+            try:
+                # Set default role for history to private
+                trans.app.security_agent.history_set_default_permissions(history, private_permissions)
+                # Set private role for all datasets
+                for hda in history.datasets:
+                    if (
+                        not hda.dataset.library_associations
+                        and not trans.app.security_agent.dataset_is_private_to_user(trans, hda.dataset)
+                        and trans.app.security_agent.can_manage_dataset(user_roles, hda.dataset)
+                    ):
+                        # If it's not private to me, and I can manage it, set fixed private permissions.
+                        trans.app.security_agent.set_all_dataset_permissions(hda.dataset, private_permissions)
+            except Exception:
```

so then we can also remove this. if we ever hit this we would lie to the user about the success of the operation 😆

---

(and it would also swallow any underlying message exception ... not that there are any now, but better to avoid this problem now)

---

Ahh alright! makes sense!